src/plugins/ResizeReorder/ResizeReorderCell.js

```diff
@@ -0,0 +1,402 @@
+     */
+    scrollToX: PropTypes.func,
+
+    /**
+     * Whether the cells belongs to the fixed group
+     */
+    isFixed: PropTypes.bool,
+
+    /**
+     * Object consisting of keys and widths of the columns
+     * in the current cell group.
+     */
+    cellGroupColumnWidths: PropTypes.object,
+
+    /**
+     * The minimum width of the column.
+     */
+    minWidth: PropTypes.number,
+
+    /**
+     * The maximum width of the column.
+     */
+    maxWidth: PropTypes.number,
+  };
+
+  initialState = {
+    isColumnReordering: false,
+    displacement: 0,
+    reorderingDisplacement: 0
+  }
+
+  state = {...this.initialState}
+  
+  clearState = () => {
+    this.setState(
+      this.initialState
+    );
+  }
+
+  startColumnReorder = (reorderData) => {
+    let { columnKey, left, scrollStart, width } = reorderData;
+    const isFixed = this.props.isFixed;
+  
+    this.updateState({
+      isColumnReordering: true,
+      columnReorderingData: {
+        cancelReorder: false,
+        dragDistance: 0,
+        isFixed: isFixed,
+        scrollStart: scrollStart,
+        columnKey: columnKey,
+        columnWidth: width,
+        originalLeft: left,
+        columnBefore: undefined,
+        columnAfter: undefined
+      }
+    });
+  }
+
+  /**
+   * TODO (sharma)
+   *
+   * scrollX gets updated while scrolling and reordering together.
+   * This causes all the components to re-render from the FixedDataTable
+   * to the ResizeReorder Cell and makes scrolling and reordering very slow.
+   *
+   * scrollX, maxScrollX, availableScrollWidth should be handled differently
+   * to speed it up.
+   *   */
+  moveColumnReorder = (deltaX) => {
+    const { isFixed, originalLeft, scrollStart } = this.state.columnReorderingData;
+    let { maxScrollX, scrollX } = this.props;
+    if (!isFixed) {
+      // Relative dragX position on scroll
+      const dragX = originalLeft - scrollStart + deltaX;
+      const availableScrollWidth = this.props.availableScrollWidth;
+      deltaX += scrollX - scrollStart;
+  
+      // Scroll the table left or right if we drag near the edges of the table
+      if (dragX > availableScrollWidth - DRAG_SCROLL_BUFFER) {
+        scrollX = Math.min(scrollX + DRAG_SCROLL_SPEED, maxScrollX);
+      } else if (dragX <= DRAG_SCROLL_BUFFER) {
+        scrollX = Math.max(scrollX - DRAG_SCROLL_SPEED, 0);
+      }
+      this.props.scrollToX(scrollX);
+    }
+  
+    // NOTE (jordan) Need to clone this object when use pureRendering
+    var reorderingData = {...this.state.columnReorderingData}
+    reorderingData.dragDistance = deltaX;
+    reorderingData.columnBefore = undefined;
+    reorderingData.columnAfter = undefined;
+    this.updateState({
+      columnReorderingData: reorderingData,
+      isColumnReordering: true
+    });
+    
+  }
+
+  stopColumnReorder = () => {
+    this.updateState({
+      isColumnReordering: false,
+      columnReorderingData: {}
+    });
+  }
+
+  updateState = (stateChanges) => {
```

Comment:
Instead of putting all this state calculation to the cell component itself, maybe it makes sense to move it to the Reorder component.
We could make use of a render prop to then avoid state syncs such that Reorder component simply renders the entire cell after translating through local state, and the cell never bothers about reordering details.

Reply:
Which component does "reorder component" refer to?  The ReorderHandle?